src/ray/raylet/node_manager.cc

```diff
@@ -1259,31 +1274,20 @@ void NodeManager::ProcessSetResourceRequest(
     return;
   }
 
-  // Add the new resource to a skeleton ClientTableData object
-  ClientTableData data;
-  gcs_client_->client_table().GetClient(client_id, data);
-  // Replace the resource vectors with the resource deltas from the message.
-  // RES_CREATEUPDATE and RES_DELETE entries in the ClientTable track changes (deltas) in
-  // the resources
-  data.add_resources_total_label(resource_name);
-  data.add_resources_total_capacity(capacity);
-  // Set the correct flag for entry_type
+  // Submit to the client table. This calls the ResourceCreateUpdated or ResourceDeleted
+  // callback, which updates cluster_resource_map_.
   if (is_deletion) {
-    data.set_entry_type(ClientTableData::RES_DELETE);
+    RAY_CHECK_OK(gcs_client_->resource_table().RemoveEntries(JobID::Nil(), client_id,
+                                                             {resource_name}, nullptr));
   } else {
-    data.set_entry_type(ClientTableData::RES_CREATEUPDATE);
-  }
-
-  // Submit to the client table. This calls the ResourceCreateUpdated callback, which
-  // updates cluster_resource_map_.
-  std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
-  if (not worker) {
-    worker = worker_pool_.GetRegisteredDriver(client);
+    std::unordered_map<std::string, std::shared_ptr<gcs::RayResource>> data_map;
+    auto ray_resource = std::make_shared<gcs::RayResource>();
+    ray_resource->set_resource_name(resource_name);
```

Comment:
I think the `resource_name` is redundant. Because the resource table is a Hash. The key is already the resource name.

Reply:
yes, we can have a `ResourceTableData` that only has a double field.

Reply:
Done.